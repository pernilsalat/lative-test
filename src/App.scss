html,
body {
    font-family: 'Verdana', sans-serif;
    font-size: 1rem;
    background-color: #fafafa;
    color: #666666;
    height: 100%;
}

#root, .app {
    height: 100%;
}

.formPanel {
    height: 20rem;
}

.header {
    text-align: center;
    font-size: 3rem;
    color: #8aa5d1;
    margin-top: 3rem;
    margin-bottom: 3rem;}

.logo {
    width: 9rem;
    height: 3rem;
    position: relative;
    left: -1rem;
}

.results {
    background-color: #3B9FC1;
    height: calc(100% - 300px);
}
