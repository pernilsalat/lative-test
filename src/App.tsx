/* Libraries */
import React, { Dispatch, SetStateAction, useMemo, useState } from 'react';

/* Components */
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';

/* Styles */
import './App.scss';
import { useFetchMesures } from './hooks/useFetchMesures';
import { CardData, FiltersForm, Measurements, ResponseState } from './types';
import { Filters, initialForm } from './components/filters';
import { Card } from './components/card';
import { postProcessMeasures } from './utils';
import { Grid } from './components/grid';

function App(): JSX.Element {
  const [form, setForm]: [FiltersForm, Dispatch<SetStateAction<FiltersForm>>] =
    useState<FiltersForm>(initialForm);

  const { data, loading }: ResponseState<Measurements> = useFetchMesures(form);

  const cardsData: CardData[] = useMemo<CardData[]>(
    () => postProcessMeasures(data || [], form.measures),
    [data]
  );
  const asd = data || [];
  console.log(asd[0], asd[asd.length/2]); // eslint-disable-line
  return (
    <div className='app'>
      <div className='formPanel'>
        <Container className='p-3'>
          <Nav>
            <Nav.Item className='logo'>
              <img src='/images/lative-logo.svg' alt='Lative Software' />
            </Nav.Item>
          </Nav>
          <h1 className='header'>Growth Ranking of U.S. States</h1>
          <Filters onChange={setForm} />
        </Container>
      </div>
      <div className='results p-3 overflow-auto d-flex justify-content-center'>
        {loading && <div className='text-white'>loading...</div>}
        {!loading && (
          <Grid<CardData> data={cardsData}>{(cardData: CardData) => <Card {...cardData} />}</Grid>
        )}
      </div>
    </div>
  );
}

export default App;
