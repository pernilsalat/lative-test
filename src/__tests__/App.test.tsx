/* Libraries */
import { render, screen } from '@testing-library/react';

/* Components */
import React from 'react';
import App from '../App';

test('renders learn react link', () => {
  // Arrange
  render(<App />);

  // Act
  const linkElement = screen.getByText(/Growth Ranking of U.S. States/i);

  // Assert
  expect(linkElement).toBeInTheDocument();
});
