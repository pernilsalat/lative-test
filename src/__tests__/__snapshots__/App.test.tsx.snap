// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component case loading should not render the grid 1`] = `HTMLCollection []`;

exports[`App component case loading should render the loader 1`] = `
<div
  class="loader text-white fs-2"
>
  loading...
</div>
`;
